app/controllers/courses_controller.rb

```diff
@@ -150,6 +184,16 @@ def edit
 
   action_auth_level :update, :instructor
   def update
+    if(@course.nil?)
+      flash[:error] = "Cannot show nil course"
+    end
+
+    if !@course.valid?
```

Comment:
Should probably show these errors in the case that @course.update fails so we don't have duplicate messages.